torchvision/transforms/v2/functional/_type_conversion.py

```diff
@@ -25,3 +29,176 @@ def to_image(inpt: Union[torch.Tensor, PIL.Image.Image, np.ndarray]) -> tv_tenso
 
 to_pil_image = _F.to_pil_image
 pil_to_tensor = _F.pil_to_tensor
+
+
+def _infer_nvcv_format(img_tensor: torch.Tensor):
+    """Infer NVCV format from tensor shape and dtype.
+
+    Args:
+        img_tensor: Tensor with shape (H, W, C) where C is number of channels.
+
+    Returns:
+        tuple: (nvcv_format, processed_tensor) where processed_tensor may have reduced dimensions
+               for single channel images.
+
+    Raises:
+        TypeError: If dtype is not supported for the given number of channels.
+        ValueError: If number of channels is not 1 or 3.
+    """
+    import nvcv  # type: ignore[import-not-found]
+
+    num_channels = img_tensor.shape[2]
+    dtype = img_tensor.dtype
+
+    # Handle single channel images
+    if num_channels == 1:
+        img_tensor = img_tensor[:, :, 0]
+        if dtype == torch.uint8:
+            return nvcv.Format.U8, img_tensor
+        elif dtype == torch.int16:
+            return nvcv.Format.S16, img_tensor
+        elif dtype == torch.int32:
+            return nvcv.Format.S32, img_tensor
+        elif dtype == torch.float32:
+            return nvcv.Format.F32, img_tensor
+        else:
+            raise TypeError(f"Unsupported dtype {dtype} for single channel image")
+
+    # Handle 3 channel images (defaults to RGB)
+    elif num_channels == 3:
+        if dtype == torch.uint8:
+            return nvcv.Format.RGB8, img_tensor
+        elif dtype == torch.float32:
+            return nvcv.Format.RGBf32, img_tensor
+        else:
+            raise TypeError(f"Unsupported dtype {dtype} for 3-channel image")
```

Comment:
Do I understand correctly that torch.float64 isn't supported by CV-CUDA? That's fine, just checking.